src/components/implementation/no_interface/scheddev/fprr/sl_timer_mod_periodic.c

```diff
@@ -0,0 +1,78 @@
+#include <sl.h>
+#include <sl_consts.h>
+#include <sl_plugins.h>
+#include <heap.h>
+
+#define SL_TIMEOUT_MOD_MAX_THDS MAX_NUM_THREADS
+
+struct wakeup_heap {
+	struct heap  h;
+	void        *data[SL_TIMEOUT_MOD_MAX_THDS];
```

Comment:
We likely want to replace all of this with rb-tree with nodes internal to the threads.  This heap is fast, but the static memory allocation is not great.

TODO: in the future.